v0.42.0 changelog improvements onward (#7828)
PennyLaneAI/pennylane: doc/releases/changelog-0.42.0.md

<h4>Experimental FTQC module</h4>

* Commutation rules for a Clifford gate set (`qml.H`, `qml.S`, `qml.CNOT`) have been added to the 
  `ftqc.pauli_tracker` module, accessible via the `commute_clifford_op` function.
  [(#7444)](https://github.com/PennyLaneAI/pennylane/pull/7444)

* Offline byproduct correction support has been added to the `ftqc` module.
  [(#7447)](https://github.com/PennyLaneAI/pennylane/pull/7447)

* The `ftqc` module `measure_arbitrary_basis`, `measure_x` and `measure_y` functions can now be 
  captured when program capture is enabled.
  [(#7219)](https://github.com/PennyLaneAI/pennylane/pull/7219)
  [(#7368)](https://github.com/PennyLaneAI/pennylane/pull/7368)

> * Functions called `pauli_to_xz`, `xz_to_pauli` and `pauli_prod` that arerelated to `xz`-encoding 

Reviewer: ```suggestion
* Functions called `pauli_to_xz`, `xz_to_pauli` and `pauli_prod` that are related to `xz`-encoding 
```

  have been added to the `ftqc` module.
  [(#7433)](https://github.com/PennyLaneAI/pennylane/pull/7433)

* A new transform called `convert_to_mbqc_formalism` has been added to the `ftqc` module to convert 
  a circuit already expressed in a limited, compatible gate set into the MBQC formalism. Circuits 
  can be converted to the relevant gate set with the `convert_to_mbqc_gateset` transform.
  [(#7355)](https://github.com/PennyLaneAI/pennylane/pull/7355)
  [(#7586)](https://github.com/PennyLaneAI/pennylane/pull/7586)

* The `RotXZX` operation has been added to the `ftqc` module to support the definition of a 

>   universal gate-set that can be translated to the MBQC formalism.

Reviewer: ```suggestion
  universal gate set that can be translated to the MBQC formalism.
```

* A new function called `qml.labs.map_to_resource_op` has been added to map PennyLane Operations to 
  their resource equivalents.
  [(#7434)](https://github.com/PennyLaneAI/pennylane/pull/7434)

* A new class called `qml.labs.Resources` has been added to store and track the quantum resources 
  from a circuit.
  [(#7406)](https://github.com/PennyLaneAI/pennylane/pull/7406)
  
* A new class called `qml.labs.CompressedResourceOp` class has been added to store information about 
  the operator type and parameters for the purposes of resource estimation.
  [(#7408)](https://github.com/PennyLaneAI/pennylane/pull/7408)

* A base class called `qml.labs.ResourceOperator` has been added which will be used to implement all 
  quantum operators for resource estimation.
  [(#7399)](https://github.com/PennyLaneAI/pennylane/pull/7399)

* A new function called `qml.labs.estimate_resources` has been added which will be used to perform 
  resource estimation on circuits, `qml.labs.ResourceOperator`, and `qml.labs.Resources` objects.
  [(#7407)](https://github.com/PennyLaneAI/pennylane/pull/7407)

* A new template called `qml.labs.ResourceOperator` has been added which will be used to perform 

>   resource estimation for non-parametric single qubit gates.

Reviewer: ```suggestion
  resource estimation for non-parametric single-qubit gates.
```

  [(#7540)](https://github.com/PennyLaneAI/pennylane/pull/7540)

* Added the `qml.labs.ResourceOperator` templates which will be used to perform resource
  estimation for parametric single qubit gates.
  [(#7541)](https://github.com/PennyLaneAI/pennylane/pull/7541)

* Added the `qml.labs.ResourceOperator` templates which will be used to perform resource
  estimation for controlled gates.
  [(#7526)](https://github.com/PennyLaneAI/pennylane/pull/7526)

* Added the `qml.labs.ResourceOperator` templates which will be used to perform resource
  estimation for symbolic operators of gates.
  [(#7584)](https://github.com/PennyLaneAI/pennylane/pull/7584)

* Added the `qml.labs.ResourceOperator` templates which will be used to perform resource

>   estimation for multi-qubit parametic gates.

Reviewer: ```suggestion
  estimation for multi-qubit parametric gates.
```

  templates which will be used to perform resource estimation for trotterization of CDF and THC Hamiltonians respectively.
  [(#7705)](https://github.com/PennyLaneAI/pennylane/pull/7705)

* Added the `qml.labs.ResourceQubitize` template which can be used to perform resource
  estimation for qubitization of THC Hamiltonian.
  [(#7730)](https://github.com/PennyLaneAI/pennylane/pull/7730)

* Added `qml.labs.resource_estimation.ResourceTrotterVibrational` and `qml.labs.resource_estimation.ResourceTrotterVibronic`
  templates which can be used to perform resource estimation for trotterization of vibrational and vibronic Hamiltonians respectively.
  [(#7720)](https://github.com/PennyLaneAI/pennylane/pull/7720)

* Added `qml.labs.ResourceOperator` templates for various algorithms required for 
  supporting compact hamiltonian development.
  [(#7725)](https://github.com/PennyLaneAI/pennylane/pull/7725)

* A new module :mod:`qml.labs.zxopt <pennylane.labs.zxopt>` provides access to the basic optimization
  passes from [pyzx](https://pyzx.readthedocs.io/en/latest/) for PennyLane circuits.
  

>     * :func:`basic_optimization <pennylane.labs.zxopt.basic_optimization>` performs peephole optimizations on the circuit and is a useful subroutine for other optimization passes.

Reviewer: Can't leave a suggestions, but below it says "the [the pyzx docs]"
